genie-web/src/main/java/com/netflix/genie/web/services/impl/AgentJobServiceImpl.java

```diff
@@ -48,25 +51,42 @@
 
     private final JobPersistenceService jobPersistenceService;
     private final JobSpecificationService jobSpecificationService;
+    private final AgentFilterService agentFilterService;
     private final MeterRegistry meterRegistry;
 
     /**
      * Constructor.
      *
      * @param jobPersistenceService   The persistence service to use
      * @param jobSpecificationService The specification service to use
+     * @param agentFilterService      The agent filter service to use
      * @param meterRegistry           The metrics registry to use
      */
     public AgentJobServiceImpl(
         final JobPersistenceService jobPersistenceService,
         final JobSpecificationService jobSpecificationService,
+        final AgentFilterService agentFilterService,
         final MeterRegistry meterRegistry
     ) {
         this.jobPersistenceService = jobPersistenceService;
         this.jobSpecificationService = jobSpecificationService;
+        this.agentFilterService = agentFilterService;
         this.meterRegistry = meterRegistry;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handshake(
+        @Valid final AgentClientMetadata agentClientMetadata
+    ) throws GenieAgentRejectedException {
```

Comment:
I think it would be good in here to add some metrics for success/failure/reason

Reply:
Done